geopandas/tests/test_geodataframe.py

```diff
@@ -393,6 +393,24 @@ def test_to_file_schema(self):
 
         assert result_schema == schema
 
+    def test_to_file_crs(self):
+        """
+        Ensure that the file is written according to the crs
+        if it is specified
+
+        """
+        tempfilename = os.path.join(self.tempdir, 'crs.shp')
+        self.df.to_file(tempfilename, crs=self.df.crs)
```

Comment:
Another possibility to test this is to actually pass a different crs than `df.crs` (can be a simple dict, that is easier to test), to make sure this is actually overwriting the `df.crs` (the current test isn't actually testing that, if I understand correctly?)